src/Adapter/Currency/CurrencyProvider.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Currency;
+
+use Currency;
+use PrestaShop\PrestaShop\Adapter\AbstractObjectModelProvider;
+use PrestaShop\PrestaShop\Core\Domain\Currency\Exception\CurrencyNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Currency\ValueObject\CurrencyId;
+use PrestaShop\PrestaShop\Core\Exception\CoreException;
+
+/**
+ * Provides existing Currency
+ */
+class CurrencyProvider extends AbstractObjectModelProvider
+{
+    /**
+     * @param CurrencyId $currencyId
+     *
+     * @return Currency
+     *
+     * @throws CoreException
+     * @throws CurrencyNotFoundException
+     */
+    public function get(CurrencyId $currencyId): Currency
+    {
+        /** @var Currency $currency */
+        $currency = $this->getObjectModel(
+            $currencyId->getValue(),
+            'currency',
```

Comment:
```suggestion
            Currency::class,
```

Reply:
not sure if its ok to provide the class namespace. As you mentioned above, It expects database table name not the class name :thinking: It will work now, but only by a coincidence. For example it would not work with CustomizationField I suppose. IMO we shouldn't provide such example just because it works now and looks better in this case, but won't work in another :thinking:

Reply:
in get() method I used the objModel::class, because parent initializes that class, its not the same here

Reply:
Mmm, I'm not sure the `assertObjectModelExists` uses the table name so it's logical to provide it via a string

But here you are using `getObjectModel` which purpose is to instanciate an object model based on the provided class, see:

https://github.com/PrestaShop/PrestaShop/blob/develop/src/Adapter/AbstractObjectModelProvider.php#L52

So it needs the **class** name not the **table** name, using `Currency::class` is totally appropriate And that's what you've done in all other providers by the way

Reply:
indeed you are right, I was sure that this is the `assertCurrencyExists` method, sorry.